src/System/Validation/ValidatedNotNullAttribute.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+
+namespace Microsoft.Omex.System.Validation
+{
+	/// <summary>
+	/// Indicates to Code analysis tools that a method validates a particular parameter to not be null.
+	/// 
+	/// This can for instance be used to suppress VS design warning CA1062 (validate public method parameters before use) by applying to
+	/// external validation libraries, such as the `Code` class in this project.

```

Comment:
>`Code` [](start = 48, length = 6)

Very small issue: it's preferable to include this as <see cref="Code"/> as the outputted documentation will be better formatted. Same for a few instances below. Where cref isn't appropriate, you can use <c>.toUpper</c> for instance.